refactor: change TensorRT-LLM metrics prefix from trtllm: to trtllm_

keivenchang opened the pull request:
#### Overview:

Changed TensorRT-LLM metrics prefix from trtllm: to trtllm_ and added logic to prevent double-prefixing.

#### Details:

- Changed prefix from trtllm: to trtllm_ in metrics registration
- Added check to prevent adding prefix if it already exists
- Updated documentation and tests to use new prefix format

#### Where should the reviewer start?

components/src/dynamo/common/utils/prometheus.py

#### Related Issues:

DIS-1008

/coderabbit profile chill

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

## Release Notes

* **Bug Fixes**
  * Fixed metric prefix handling to prevent double-prefixing when metrics already include the designated prefix.

* **Documentation**
  * Updated Prometheus metric documentation and examples to reflect new underscore-based naming convention for TensorRT-LLM backend metrics (e.g., `trtllm_request_latency_seconds` instead of `trtllm:request_latency_seconds`).

* **Other Changes**
  * Metric naming convention updated from colon-delimited to underscore-delimited format across all relevant APIs.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

keivenchang committed:
refactor: change TensorRT-LLM metrics prefix from trtllm: to trtllm_

Signed-off-by: Keiven Chang <keivenchang@users.noreply.github.com>

ajcasagrande commented on components/src/dynamo/common/utils/prometheus.py:
```diff
@@ -169,16 +169,37 @@ def get_prometheus_expfmt(
+                            ):
+                                metric_name_part = metric_name_part[
+                                    len(metric_prefix_filter) :
+                                ]
```
option to consider
```suggestion
                            if metric_prefix_filter:
                                metric_name_part = metric_name_part.removeprefix(
                                    metric_prefix_filter
                                )
```

ajcasagrande reviewed (approved):
LGTM

nv-yilinf reviewed (approved):
LGTM. Thanks!

keivenchang committed:
refactor using startswith()

Signed-off-by: Keiven Chang <keivenchang@users.noreply.github.com>

keivenchang committed:
Merge main into keivenchang/DIS-1008__trtllm-prefix-fixes

Signed-off-by: Keiven Chang <keivenchang@users.noreply.github.com>

keivenchang committed:
Improve test validation with flexible keyword matching

- Change validation from AND to OR logic for expected keywords
- Accept any of: AI, knock, joke, think, artificial, intelligence
- Add match count logging (e.g., 'Found 4/6 expected keywords')
- Update prompt to 'Tell me a knock knock joke about AI'
- Set max_tokens=1000, temperature=0.0 for consistent responses

Signed-off-by: Keiven Chang <keivenchang@users.noreply.github.com>
